plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/launching/LaunchHelper.java

```diff
@@ -134,29 +130,16 @@ private IServer createServer(IModule[] modules, SubMonitor progress) throws Core
     return serverWorkingCopy.save(false, progress.newChild(2));
   }
 
-  protected void launch(final IServer server, final String launchMode, SubMonitor progress) {
+  protected void launch(IServer server, String launchMode, SubMonitor progress)
```

Comment:
Will you add `@VisbileForTesting`? I verified it can be `private` if not for a test.

Reply:
Sure, especially to see if this TimeoutException on Kokoro is repeatable :-/

```
2018-01-16T17:42:11.634017514Z  testDebugLaunch(com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest)  Time elapsed: 39.816 sec  <<< ERROR!
2018-01-16T17:42:11.634017514Z  org.eclipse.swtbot.swt.finder.widgets.TimeoutException: Timeout after: 30000 ms.: The widget with mnemonic 'Remove All Terminated' was not enabled.
2018-01-16T17:42:11.634017514Z  at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:522)
2018-01-16T17:42:11.634017514Z  at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:496)
2018-01-16T17:42:11.634017514Z  at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:484)
2018-01-16T17:42:11.634017514Z  at org.eclipse.swtbot.swt.finder.widgets.AbstractSWTBot.waitForEnabled(AbstractSWTBot.java:694)
2018-01-16T17:42:11.634017514Z  at org.eclipse.swtbot.swt.finder.widgets.SWTBotMenu.click(SWTBotMenu.java:73)
2018-01-16T17:42:11.634017514Z  at com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest.testDebugLaunch(DebugNativeAppEngineStandardProjectTest.java:110)
```

Reply:
This unguarded use of _Remove All Terminated_ seems rather error-prone:
```
    // avoid any stray processes that may be lying around
    launchTree.contextMenu("Remove All Terminated").click();
```
This code assumes that there must be some stray processes; surely it should check if it's enabled before trying to click?